src/app/services/species.service.spec.ts

```diff
@@ -29,9 +29,13 @@ describe('SpeciesService', () => {
   });
 
   it('should get a species by id', () => {
-    httpClientSpy.get.and.returnValue(of(species.results[0]));
-    service.getById(0).subscribe(data => {
-      expect(data).toEqual(species.results[0]);
+    let indexURL = species.results[0].url.slice(0,-1);
```

Comment:
This is a lot of work to use a number you already know the value of. Simply fake the id of the item in question and use it as an argument to `service.getById`.

Remember: you're not testing the inner logic of functions. Youre testing state and behavior.

Reply:
We fixed this issue with the new commit 'e2f007e'